buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -1673,6 +1703,20 @@ class BeamModulePlugin implements Plugin<Project> {
         }
         return argList.join(' ')
       }
+
+      project.ext.toxTask = { name, tox_env ->
+        project.tasks.create(name) {
+          dependsOn = ['sdist']
+          doLast {
+            project.exec {
+              executable 'sh'
+              args '-c', ". ${project.ext.envdir}/bin/activate && ${pythonRootDir}/scripts/run_tox.sh $tox_env ${project.buildDir}/apache-beam.tar.gz"
```

Comment:
Does this step require both sources in a separated directory and a tarball archive produced by sdist? Why not just one of them?

Reply:
I only pass tarball to the script. `$tox_env` is the name of environment that we want to run (like `py27-lint, py3-gcp`).